### PR TITLE
NCBI taxonomy data import

### PR DESCRIPTION
**Pull request by laughedelic**
Again, look at bio4j/bio4j.

### CODE REVIEW

**Comment by laughedelic** [comment]
@eparejatobes review it, please.

File: src/main/scala/ncbiTaxonomy/ncbiTaxonomy.scala [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    }
+  )
+
+  val parents = GraphProcess.generically[V,E] (graph,
+    (node: TaxonNode, g: G) => {
+
+      val srcOpt = g.taxon.id.index.find(node.taxID).asScala
+      val tgtOpt = g.taxon.id.index.find(node.parentTaxID).asScala
+
+      val edges = (srcOpt zip tgtOpt).map { case (s, t) =>
```
Comment (eparejatobes) [inline src/main/scala/ncbiTaxonomy/ncbiTaxonomy.scala]: better edge?

File: src/main/scala/ncbiTaxonomy/ncbiTaxonomy.scala [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      (graph, edges)
+    }
+  )
+
+  val names = GraphProcess.generically[V,E] (graph,
+    (taxName: ScientificName, g: G) => {
+
+      val taxon = g.taxon.id.index.find(taxName.taxID).asScala
+
+      val vertices = taxon.map {
```
Comment (eparejatobes) [inline src/main/scala/ncbiTaxonomy/ncbiTaxonomy.scala]: same, vertex

File: src/main/scala/ncbiTaxonomy/ncbiTaxonomy.scala [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        _.set(g.taxon.name, taxName.scientificName)
+      }
+
+      (graph, vertices)
+    }
+  )
+
+  private def findNode(g: G, id: String): Option[G#Taxon] =
```
Comment (eparejatobes) [inline src/main/scala/ncbiTaxonomy/ncbiTaxonomy.scala]: either use it or remove it

**Review by eparejatobes — COMMENTED** [review]
done, have you tested it with the dmp files?

File: src/main/scala/ncbiTaxonomy/ncbiTaxonomy.scala (hunk 1 above)
Comment (laughedelic) [reply src/main/scala/ncbiTaxonomy/ncbiTaxonomy.scala]: Yes. And I just realised that I shouldn't create it for the root. It is its own parent 😱

File: src/main/scala/ncbiTaxonomy/ncbiTaxonomy.scala (hunk 2 above)
Comment (laughedelic) [reply src/main/scala/ncbiTaxonomy/ncbiTaxonomy.scala]: Here its type is `Option`, but I guess we don't do anything with those returned values, do we?

**Comment by laughedelic** [comment]
I'm testing it now and going to commit a couple of little fixes.

**Comment by laughedelic** [comment]
Done 👌 I added some tests, but ran it only for the first 1000 nodes (it just takes time (and memory)).

**Comment by eparejatobes** [comment]
LGTM from me

**Pull request merged**